src/templates/blog-index.js

```diff
@@ -55,7 +55,7 @@ BlogIndexPage.propTypes = {
 }
 
 export const pageQuery = graphql`
-  query BlogIndex
+  query BlogIndex($id: String!)
```

Comment:
This cause the issue: we receive the wrong id frontmatter for this reason!
See also below...